### ISSUE TITLE
Scaffolder plugin review mask does not work as documented

### ISSUE THREAD

**Issue by ivd-git** [backstage/backstage#11687]
## Expected Behavior

To hide or mask this data, you can set the `review` properties of `ui:backstage`. From the documentation:

```yaml
- title: Hide or mask values
  properties:
    masked:
      title: Masked
      type: string
      ui:backstage:
        review:
          mask: '<some-value-to-show>' # will print '<some-value-to-show>' as value for property 'Masked' on Review Step
```

## Actual Behavior

But in reality the `masked` field is hidden

## Steps to Reproduce

1. In the `MultistepJsonForm.test.tsx` file [on line 44] for `stepsMock.schema.properties.masked.'ui:backstage'.review` remove the line `show: true`
2. Run the tests from `MultistepJsonForm.test.tsx`, it fails on `expect(reviewData.masked).toBe('******');`

## Context

The mismatch between documentation and actual behavior cost me some time. Nothing serious, but it would be nice to make sure both are aligned.

## Your Environment

Not relevant 

## Proposed solution:

Either update the documentation to indicate that `show: true` is required when defining a mask, or update the code so that it behaves according to the current documentation.

**Comment by benjdlambert** [comment]
@ivd-git would you like to make the required documentation change for now? maybe we can revisit this API at a later date too to make it behave slightly different.

**Comment by ivd-git** [comment]
This can be closed because PR #11732 was merged.
